packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/endpoint_client/endpoint_client.dart

```diff
@@ -20,12 +18,15 @@ import 'package:built_collection/built_collection.dart';
 /// {@endtemplate}
 class EndpointClient {
   /// {@macro amplify_analytics_pinpoint_dart.endpoint_client}
-  EndpointClient._({
+  EndpointClient({
     required String pinpointAppId,
     required PinpointClient pinpointClient,
+    required EndpointInfoStoreManager endpointInfoStoreManager,
```

Comment:
We could technically required fixedEndpointId and globalFieldsManager as DI, but I decided to still required EndpointInfoStoreManager as that class is meant to enforce a consistent storage/retrieval behavior for those values.